### PR TITLE
add trace benchmark

### PR DESCRIPTION
**Pull request by vthacker**

### CODE REVIEW

File: benchmarks/src/main/java/com/slack/kaldb/IndexTraceLogBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+import org.apache.lucene.store.Directory;
+import org.openjdk.jmh.annotations.*;
+
+@State(Scope.Thread)
+public class IndexTraceLogBenchmark {
+
+  private static final String INDEX_NAME = "spans";
+
+  private Random random;
+  private final Duration commitInterval = Duration.ofSeconds(5 * 60);
```
Comment (bryanlb) [inline benchmarks/src/main/java/com/slack/kaldb/IndexTraceLogBenchmark.java:34]: `Duration.ofMinutes()`?

File: benchmarks/src/main/java/com/slack/kaldb/IndexTraceLogBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  //        skipCount++;
+  //      }
+  //      System.out.println("Unable to parse " + skipCount + " lines");
+  //    }
+  //  }
+
+  @Setup(Level.Iteration)
+  public void createIndexer() throws Exception {
+
+    //        reader = Files.newBufferedReader(Path.of("trace_logs.txt"),
```
Comment (bryanlb) [inline benchmarks/src/main/java/com/slack/kaldb/IndexTraceLogBenchmark.java:94]: Left over?

File: benchmarks/src/main/java/com/slack/kaldb/IndexTraceLogBenchmark.java [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+  @TearDown(Level.Iteration)
+  public void tearDown() throws IOException {
+    Directory directory = logStore.getIndexWriter().getDirectory();
+    String[] segmentFiles = directory.listAll();
+    long indexedBytes = 0;
+    for (String segmentFile : segmentFiles) {
+      indexedBytes += directory.fileLength(segmentFile);
+    }
+    System.out.println(
```
Comment (bryanlb) [inline benchmarks/src/main/java/com/slack/kaldb/IndexTraceLogBenchmark.java:121]: Are we using the SLF4J logger for the tests? If so we should probably use it here. If not, why do we have a [benchmark specific SLF4J config](https://github.com/slackhq/kaldb/blob/master/benchmarks/log4j2.xml)?

**Review by bryanlb — DISMISSED** [review]

File: benchmarks/src/main/java/com/slack/kaldb/IndexTraceLogBenchmark.java (hunk 3 above)
Comment (vthacker) [reply benchmarks/src/main/java/com/slack/kaldb/IndexTraceLogBenchmark.java:73]: very good point! haha I had forgotten

```
[INFO ] 2021-09-22 13:28:46.221 [com.slack.kaldb.IndexTraceLogBenchmark.measureTraceLogIndexingSlingshotMode-jmh-worker-1]  IndexTraceLogBenchmark - Indexed = 100 Skipped = 0 Index size = 47 KB

```

using the logger and output gets printed to console

File: benchmarks/src/main/java/com/slack/kaldb/IndexAPILog.java [hunk 4]
```diff
@@ -69,15 +73,15 @@ public void tearDown() throws IOException {
...
+      LOG.info(
           "Indexed = "
               + indexCount
               + " Skipped = "
               + skipCount
               + " Index size = "
               + FileUtils.byteCountToDisplaySize(indexedBytes));
     } else {
-      System.out.println(
+      LOG.info((
```
Comment (bryanlb) [inline benchmarks/src/main/java/com/slack/kaldb/IndexAPILog.java:84]: Looks like this is causing compilation to fail

**Review by bryanlb — CHANGES_REQUESTED** [review]
Looks like it's failing compilation

**Pull request closed without merging**